ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/contentassist/TypePackageCompletionProcessor.java

```diff
@@ -134,11 +134,11 @@ public void accept(CompletionProposal proposal) {
 						pName = completion.substring(0, period);
 					}
 					Image image = isInterface ? PDEPluginImages.get(PDEPluginImages.OBJ_DESC_GENERATE_INTERFACE) : PDEPluginImages.get(PDEPluginImages.OBJ_DESC_GENERATE_CLASS);
-					String label = cName;
+					StringBuilder label = new StringBuilder().append(cName);
```

Comment:
I don't think this is an improvement in terms of performance or readability.

https://openjdk.org/jeps/280 as well as the optimized code that ECJ emits for these trivial cases (string + string + string)  outweigh the whatever was intended here.